packages/components/src/settings/index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { Form } from '../index';
+import {
+	SettingText,
+	SettingPassword,
+	SettingCheckbox,
+	SettingSelect,
+} from './types';
+
+const typeMap = {
```

Comment:
Nice!  Wondering if we should have any type of error handling if a type is invalid or default to text.

Reply:
Good call, I'll throw in `text` as default and a warning if it's an unsupported type.